### ISSUE TITLE
deepin-screenshot window not covering whole screen

### ISSUE THREAD

**Issue by hexchain** [linuxdeepin/developer-center#311]
Please see https://github.com/linuxdeepin/deepin-screenshot/issues/11.

I've reproduced this on AwesomeWM. `deepin-screenshot` windows are floating, as defined in WM rules.

Seems the problem can be fixed by setting the `Qt::Tool` flag:

```diff
diff --git a/screenshot.cpp b/screenshot.cpp
index b5e3d4e..813a914 100644
--- a/screenshot.cpp
+++ b/screenshot.cpp
@@ -7,7 +7,7 @@ Screenshot::Screenshot(QWidget *parent)
     : QMainWindow(parent)
 {
     setWindowFlags(Qt::X11BypassWindowManagerHint | Qt::WindowStaysOnTopHint |
-                   Qt::FramelessWindowHint);
+                   Qt::FramelessWindowHint | Qt::Tool);
     setAttribute(Qt::WA_TranslucentBackground);
 
     m_eventContainer = new EventContainer(this);
```

I'm no expert on Qt or X11, so not sure if `Qt::Tool` fits.

**Comment by sbwtw** [comment]
the description of `Qt::X11BypassWindowManagerHint` is:
Bypass the window manager completely. This results in a **borderless** window that is not managed at all (i.e., no keyboard input unless you call QWidget::activateWindow() manually).

so it's maybe your wm' problem. @sonald 

BTW, I think `Qt::FramelessWindowHint` is not effect when `Qt::X11BypassWindowManagerHint` is set. @pangpangpang3

**Comment by ag14774** [comment]
Is this in any way related to this: https://github.com/linuxdeepin/developer-center/issues/309 ?

**Comment by hexchain** [comment]
The current master branch is good while 4.0.3 is bad. `git bisect` shows that 5cdd7fa is the commit  that fixed this issue.

**Comment by pangpangpang3** [comment]
The problem has been fixed, the new tag is 4.0.9.
